kafka_consumer/datadog_checks/kafka_consumer/client.py

```diff
@@ -133,9 +133,9 @@ def get_highwater_offsets(self, consumer_offsets):
                     consumer_group,
                 )
                 for topic_partition_with_highwater_offset in consumer.offsets_for_times(
-                    partitions=list(topic_partitions_for_highwater_offsets)
+                    partitions=list(topic_partitions_for_highwater_offsets), timeout=self.config._request_timeout,
```

Comment:
Is there a way to make this baked into all requests without having to pass it explicitly from the config?

Reply:
I tried, but looking at the docs and the code, I don't think so, but I can be wrong:

- Docs: https://docs.confluent.io/platform/current/clients/confluent-kafka-python/html/index.html#confluent_kafka.Consumer.offsets_for_times
- Confluent-kafka code: https://github.com/confluentinc/confluent-kafka-python/blob/6dd1e5784bcb77297015ca1e4636ef6c6ec2af34/src/confluent_kafka/src/Consumer.c#L904-L953 with cfl_timeout_ms: https://github.com/confluentinc/confluent-kafka-python/blob/master/src/confluent_kafka/src/confluent_kafka.h#L456-L460
- Librdkafka code: https://github.com/confluentinc/librdkafka/blob/49f180a36c247100dc246379ace84833b91f4038/src/rdkafka.c#L3701-L3760 with rd_timeou_init: https://github.com/confluentinc/librdkafka/blob/master/src/rdtime.h#L178-L183

Reply:
this class is in fact the abstraction, this is the client we use in the check

Reply:
This class doesn't belong to us, right? it's coming from the underlying library. I skimmed the code a bit, and I find it a bit confusing/inconsistent that we sometimes  use our own `KafkaClient` (which does pass the timeout in the background btw) but sometimes use the `Consumer` directly.

Also, I'm wondering [what this does](https://github.com/DataDog/integrations-core/blob/6f63a3d17ad0ec68c17c20df0d0f3f971f4989f3/kafka_consumer/datadog_checks/kafka_consumer/client.py#L24) 🤔

Reply:
I'm talking about the Client class, that belongs to us and is a wrapper around the confluent-kafka stuff.

Consumer and KafkaClient are two different classes that do not allow us to do the same stuff. For offset related things, we need Consumer. 

This set the timeout used when performing requests, which is set to `60000` by default

Reply:
Oh, I get it now, thanks for bearing with me! carry on!